src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -91,7 +91,6 @@ public MenuSession(String username, String domain, String appId, String locale,
                        boolean oneQuestionPerScreen, String asUser, boolean preview) throws Exception {
         String resolvedInstallReference = resolveInstallReference(appId, host, domain);
         this.session = new SerializableMenuSession(
-                UUID.randomUUID().toString(),
```

Comment:
with this being auto-generated on save we have to make sure it's not used before the save happens.

Reply:
@sravfeyn I haven't verified yet if this is an issue - just need to check where this ID is used outside the object in comparison to where save happens

Reply:
@sravfeyn have you looked into this yet?

Reply:
Yeah, I reviewed in various places where `SerializableMenuSession` is being instantiated or `saveSession` is called and I haven't find any places where the ID is being used.

Reply:
and is the model always saved before it is serialized for the response?